### PR TITLE
[Android] Refactor DB read code to use RxJava

### PR DESCRIPTION
**Pull request by sigmabeta**
The first of many "architect the Android code to kill duplicate code" PRs.

In particular, this code gets rid of Loaders in the UI, which are how Google wants you to do asynchronous DB access. It's a lot of code though, and it's really unreliable. RxJava is much cleaner and easier to maintain, and we'll be able to use it for a lot of the other async operations we need.

### CODE REVIEW

**Comment by delroth** [comment]
Please get an LGTM from HdkR. I don't have an opinion on this, the only thing I checked is that it won't have an impact on the 5.0 release :)

@dolphin-emu-bot allowmerge

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/PlatformGamesFragment.java [hunk 1]
```diff
@@ -87,32 +89,34 @@ public void refreshScreenshotAtPosition(int position)
...
+	public void loadGames() {
+		Log.d("DolphinEmu", "[PlatformGamesFragment] " + mPlatform + ": Loading games...");
+
+		GameDatabase databaseHelper = DolphinApplication.databaseHelper;
+
+		databaseHelper.getGamesForPlatform(mPlatform)
+				.subscribeOn(Schedulers.io())
+				.observeOn(AndroidSchedulers.mainThread())
+				.subscribe(new Action1<Cursor>()
+						   {
```
Comment (Sonicadvance1) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/PlatformGamesFragment.java]: Weird mixture of tabs and spaces in this section.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/GameDatabase.java [hunk 2]
```diff
@@ -252,4 +255,46 @@ else if (!folder.exists())
...
+			@Override
+			public void call(Subscriber<? super Cursor> subscriber)
+			{
+				Log.i("DolphinEmu", "[GameDatabase] Reading games list...");
+
+				String whereClause;
+				String [] whereArgs;
+
+				// If -1 passed in, return all games. Else, return games for one platform only.
+				if (platform >= 0) {
```
Comment (Sonicadvance1) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/GameDatabase.java]: curly brace on the next line.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/GameDatabase.java [hunk 3]
```diff
@@ -252,4 +255,49 @@ else if (!folder.exists())
...
+	{
+		return Observable.create(new Observable.OnSubscribe<Cursor>()
+		{
+			@Override
+			public void call(Subscriber<? super Cursor> subscriber)
+			{
+				Log.i("DolphinEmu", "[GameDatabase] Reading games list...");
+
+				String whereClause;
+				String[] whereArgs;
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/GameDatabase.java]: These can just be initialized to null, and have the else statement removed.

**Comment by lioncash** [comment]
LGTM other than that

**Pull request merged**